### PR TITLE
[EditorSpatialGizmo] Fix error in intersect_ray

### LINKED ISSUE #28737: Error `look_at() failed` when look from top to the scene
**Godot version:**
3.2.dev.custom_build. df18c8c

**OS/device including version:**
Ubuntu 19.04

**Issue description:**
When I look to the scene from top, then this error appear:
```
ERROR: look_at: Up vector and direction between node origin and target are aligned, look_at() failed
   At: scene/3d/spatial.cpp:687.
```

**Steps to reproduce:**
https://streamable.com/zr5t1

### PR DESCRIPTION
**Pull request by guilhermefelipecgs**
Fix #28737

### CODE REVIEW

File: editor/spatial_editor_gizmos.cpp [hunk 1]
```diff
@@ -569,20 +569,13 @@ bool EditorSpatialGizmo::intersect_ray(Camera *p_camera, const Point2 &p_point,
 
 		Point2 center = p_camera->unproject_position(t.origin);
 
-		Transform orig_camera_transform = p_camera->get_camera_transform();
-
-		if (orig_camera_transform.origin.distance_squared_to(t.origin) > 0.01) {
```
Comment (JFonS) [inline editor/spatial_editor_gizmos.cpp]: The `look_at` call is there for a reason, not having it cuses errors when selecting icons that are on the border of the viewport. I think a better solution would be to add another guard to this if statement:
`ABS(orig_camera_transform.basis.get_axis(2).dot(Vector3(0,1,0))) < 0.99`

So `look_at` won't be called if the camera is already aligned with the target icon.

Comment (guilhermefelipecgs) [reply editor/spatial_editor_gizmos.cpp]: I was wondering what that code was doing.

Comment (guilhermefelipecgs) [reply editor/spatial_editor_gizmos.cpp]: Thanks.

**Review by JFonS — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**